torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -82,7 +82,7 @@ bool FunctionParameter::check(PyObject* obj) {
       // if a size is specified (e.g. IntList[2]) we also allow passing a single int
       return size > 0 && THPUtils_checkLong(obj);
     }
-    case ParameterType::GENERATOR: return false;
+    case ParameterType::GENERATOR: return THPGenerator_Check(obj);
```

Comment:
Is there any point in doing this check if it can't be used anyway? `None`s should be handled in a different pathway anyway, right?

Reply:
It's just a slightly nicer error message in the case where you actually pass in a generator.  I spent a few minutes trying to figure out why my torch.default_generator was a torch._C.Generator and not a torch.Generator before I figured out it just rejects any object and the error message is bogus.  Now it just tells you only None is supported in that case.